documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,947 @@
+operation. `AWSIotTopic` object (or, usually, an object of a children class) implements `onSuccess` and `onFailure` methods.
+One of these methods will be called after the operation succeeds or fails.
+
+The v2 SDK provides only asynchronous non-blocking API. First, you need to create an [UnsubscribePacket](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/packets/UnsubscribePacket.html)
+object with the help of [UnsubscribePacketBuilder](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/packets/UnsubscribePacket.UnsubscribePacketBuilder.html).
+The [unsubscribe](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/Mqtt5Client.html#unsubscribe(software.amazon.awssdk.crt.mqtt5.packets.UnsubscribePacket))
+operation takes a description of the [UnsubscribePacket](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/packets/UnsubscribePacket.html)
+you wish to send and returns a promise that resolves successfully with the corresponding [UnsubAckPacket](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/packets/UnsubAckPacket.html)
+returned by the broker. The promise is rejected with an error if anything goes wrong before the [UnsubAckPacket](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/packets/UnsubAckPacket.html)
+is received. You should always check the reason codes of a [UnsubAckPacket](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/packets/UnsubAckPacket.html)
+completion to determine if the unsubscribe operation actually succeeded.
+
+Similar to subscribing, you can unsubscribe from multiple topics in one request by just calling [withSubscription](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/packets/UnsubscribePacket.UnsubscribePacketBuilder.html#withSubscription(java.lang.String))
+for each topic you wish to unsubscribe from.
+
+#### Example of unsubscribing in the v1 SDK
+
+```java
+// Blocking API.
+client.unsubscribe("my/topic");
+client.unsubscribe("another/topic");
+```
+
+```java
+// Non-blocking API.
+public class MyTopic extends AWSIotTopic {    
+    public MyTopic(String topic, AWSIotQos qos) {
+        super(topic, qos);
+    }
+
+    @Override
+    public void onSuccess() {
+        // Called when unsubscribing succeeds.
+    }
+    
+        @Override
+    public void onFailure() {
+        // Called when unsubscribing fails.
+    }
+}
+
+// Unsubscribe from topic.
+MyTopic myOwnTopic = new MyTopic("my/topic", AWSIotQos.QOS1);
+client.unsubscribe(myOwnTopic);
+```
+
+#### Example of unsubscribing in the v2 SDK
+
+```java
+UnsubscribePacketBuilder unsubBuilder = new UnsubscribePacketBuilder("my/topic");
+client.unsubscribe(unsubBuilder.build()).get(60, TimeUnit.SECONDS);
+```
+
+### Client stop
```

Comment:
Noting that we have a "Client stop" but not a "Client start" section. Reiterating that we should remove the throwaway line about v2 mqtt5 having start/stop and v1 having connect/disconnect and just have the differences explained through these points in the doc.